fastapi_utils/inferring_router.py

```diff
@@ -12,5 +12,7 @@ class InferringRouter(APIRouter):
 
         def add_api_route(self, path: str, endpoint: Callable[..., Any], **kwargs: Any) -> None:
             if kwargs.get("response_model") is None:
-                kwargs["response_model"] = get_type_hints(endpoint).get("return")
+                return_hint = get_type_hints(endpoint).get("return")
+                if return_hint not in (type(None), NoReturn):  # noqa: E721
```

Comment:
I got flake8 to accept this using

`if return_hint not in (None.__class__, NoReturn):`